src/transformers/configuration_utils.py

```diff
@@ -855,6 +855,9 @@ def to_diff_dict(self) -> Dict[str, Any]:
 
         self.dict_torch_dtype_to_str(serializable_config_dict)
 
+        if "_flash_attn_2_enabled" in serializable_config_dict:
+            del serializable_config_dict["_flash_attn_2_enabled"]
```

Comment:
If the attribute was loaded ( so someone trained / used it) we need to set it to `False` if the checks don't pass. Otherwise, if someone put's it in the kwargs, and checks don't pass we must error out.